osmosis/src/storeProvider.js

```diff
@@ -1,4 +1,11 @@
 export const StoreProvider = (storeProviders, wrappedComponent) => {
-  storeProviders.reverse().forEach(provider => (wrappedComponent = provider(wrappedComponent)));
-  return wrappedComponent;
+  if(!Array.isArray(storeProviders)) throw new Error('StoreProvider requires an array of wrapper functions')
+  if(!wrappedComponent) throw new Error('StoreProvider requires the root component of your app')
```

Comment:
@jkhusanov this isnt exactly true....while it will be common to wrap the root component, you can technically wrap any component to supply state management from a store to any subsection of the app. We should probably tweak this to something like the below
```
StoreProvider requires a component to wrap
```

Reply:
I was trying to follow the docs wording https://github.com/shipt/osmosis/tree/master/osmosis#usage
But yeah we can make the message more generic.